query/src/main/java/org/infinispan/query/internal/QueryBlockHoundIntegration.java

```diff
@@ -12,5 +13,6 @@ public class QueryBlockHoundIntegration implements BlockHoundIntegration {
    public void applyTo(BlockHound.Builder builder) {
       // Loading a service may require opening a file from classpath
       builder.allowBlockingCallsInside(NamedSPILoader.class.getName(), "reload");
+      builder.allowBlockingCallsInside(DistributedIndexerLock.class.getName(), "lock");
```

Comment:
Is this really for `DistributedIndexerLock#getLock`?

Reply:
actually is for `ClusteredLock#tryLock`

Reply:
Actually it seems that both the methods (getLock and tryLock) can lead to the execution of `java.util.concurrent.locks.ReentrantLock#lock` for instance. I'm sending the two stack traces.

Reply:
At this point I would add the entire `DistributedIndexerLock#lock` that is supposed to cover both. WDYT?

Reply:
Ah! it seems what I did it.... :P